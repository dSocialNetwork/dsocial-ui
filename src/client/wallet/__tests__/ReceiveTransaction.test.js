import React from 'react';
import { shallow } from 'enzyme';
import ReceiveTransaction from '../ReceiveTransaction';

describe('(Component) ReceiveTransaction', () => {
  describe('with default prop values', () => {
    it('renders and matches snapshot', () => {
      const props = {
        from: 'holadweb',
        memo: 'Test Memo',
        amount: <span>{'0 BEX'}</span>,
        timestamp: '0',
      };
      const wrapper = shallow(<ReceiveTransaction {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
