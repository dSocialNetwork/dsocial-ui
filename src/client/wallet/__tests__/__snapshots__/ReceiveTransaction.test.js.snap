// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ReceiveTransaction with default prop values renders and matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReceiveTransaction
    amount={
      <span>
        0 BEX
      </span>
    }
    from="holadweb"
    memo="Test Memo"
    timestamp="0"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="UserWalletTransactions__avatar"
        >
          <Avatar
            size={40}
            username="holadweb"
          />
        </div>,
        <div
          className="UserWalletTransactions__content"
        >
          <div
            className="UserWalletTransactions__content-recipient"
          >
            <div>
              <FormattedMessage
                defaultMessage="Received from {username}"
                id="received_from"
                values={
                  Object {
                    "username": <Link
                      replace={false}
                      to="/@holadweb"
                    >
                      <span
                        className="username"
                      >
                        holadweb
                      </span>
                    </Link>,
                  }
                }
              />
            </div>
            <div
              className="UserWalletTransactions__received"
            >
              +
              <span>
                0 BEX
              </span>
            </div>
          </div>
          <span
            className="UserWalletTransactions__timestamp"
          >
            <BTooltip
              title={
                <span>
                  <FormattedDate
                    value="0Z"
                  />

                  <FormattedTime
                    value="0Z"
                  />
                </span>
              }
            >
              <span>
                <FormattedRelative
                  updateInterval={10000}
                  value="0Z"
                />
              </span>
            </BTooltip>
          </span>
          <span
            className="UserWalletTransactions__memo"
          >
            Test Memo
          </span>
        </div>,
      ],
      "className": "UserWalletTransactions__transaction",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Avatar
            size={40}
            username="holadweb"
          />,
          "className": "UserWalletTransactions__avatar",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": 40,
            "username": "holadweb",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="UserWalletTransactions__content-recipient"
            >
              <div>
                <FormattedMessage
                  defaultMessage="Received from {username}"
                  id="received_from"
                  values={
                    Object {
                      "username": <Link
                        replace={false}
                        to="/@holadweb"
                      >
                        <span
                          className="username"
                        >
                          holadweb
                        </span>
                      </Link>,
                    }
                  }
                />
              </div>
              <div
                className="UserWalletTransactions__received"
              >
                +
                <span>
                  0 BEX
                </span>
              </div>
            </div>,
            <span
              className="UserWalletTransactions__timestamp"
            >
              <BTooltip
                title={
                  <span>
                    <FormattedDate
                      value="0Z"
                    />

                    <FormattedTime
                      value="0Z"
                    />
                  </span>
                }
              >
                <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value="0Z"
                  />
                </span>
              </BTooltip>
            </span>,
            <span
              className="UserWalletTransactions__memo"
            >
              Test Memo
            </span>,
          ],
          "className": "UserWalletTransactions__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <FormattedMessage
                    defaultMessage="Received from {username}"
                    id="received_from"
                    values={
                      Object {
                        "username": <Link
                          replace={false}
                          to="/@holadweb"
                        >
                          <span
                            className="username"
                          >
                            holadweb
                          </span>
                        </Link>,
                      }
                    }
                  />
                </div>,
                <div
                  className="UserWalletTransactions__received"
                >
                  +
                  <span>
                    0 BEX
                  </span>
                </div>,
              ],
              "className": "UserWalletTransactions__content-recipient",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <FormattedMessage
                    defaultMessage="Received from {username}"
                    id="received_from"
                    values={
                      Object {
                        "username": <Link
                          replace={false}
                          to="/@holadweb"
                        >
                          <span
                            className="username"
                          >
                            holadweb
                          </span>
                        </Link>,
                      }
                    }
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "defaultMessage": "Received from {username}",
                    "id": "received_from",
                    "values": Object {
                      "username": <Link
                        replace={false}
                        to="/@holadweb"
                      >
                        <span
                          className="username"
                        >
                          holadweb
                        </span>
                      </Link>,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "+ ",
                    <span>
                      0 BEX
                    </span>,
                  ],
                  "className": "UserWalletTransactions__received",
                },
                "ref": null,
                "rendered": Array [
                  "+ ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "0 BEX",
                    },
                    "ref": null,
                    "rendered": "0 BEX",
                    "type": "span",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <BTooltip
                title={
                  <span>
                    <FormattedDate
                      value="0Z"
                    />

                    <FormattedTime
                      value="0Z"
                    />
                  </span>
                }
              >
                <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value="0Z"
                  />
                </span>
              </BTooltip>,
              "className": "UserWalletTransactions__timestamp",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value="0Z"
                  />
                </span>,
                "title": <span>
                  <FormattedDate
                    value="0Z"
                  />

                  <FormattedTime
                    value="0Z"
                  />
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <FormattedRelative
                    updateInterval={10000}
                    value="0Z"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "updateInterval": 10000,
                    "value": "0Z",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
              "type": [Function],
            },
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test Memo",
              "className": "UserWalletTransactions__memo",
            },
            "ref": null,
            "rendered": "Test Memo",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="UserWalletTransactions__avatar"
          >
            <Avatar
              size={40}
              username="holadweb"
            />
          </div>,
          <div
            className="UserWalletTransactions__content"
          >
            <div
              className="UserWalletTransactions__content-recipient"
            >
              <div>
                <FormattedMessage
                  defaultMessage="Received from {username}"
                  id="received_from"
                  values={
                    Object {
                      "username": <Link
                        replace={false}
                        to="/@holadweb"
                      >
                        <span
                          className="username"
                        >
                          holadweb
                        </span>
                      </Link>,
                    }
                  }
                />
              </div>
              <div
                className="UserWalletTransactions__received"
              >
                +
                <span>
                  0 BEX
                </span>
              </div>
            </div>
            <span
              className="UserWalletTransactions__timestamp"
            >
              <BTooltip
                title={
                  <span>
                    <FormattedDate
                      value="0Z"
                    />

                    <FormattedTime
                      value="0Z"
                    />
                  </span>
                }
              >
                <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value="0Z"
                  />
                </span>
              </BTooltip>
            </span>
            <span
              className="UserWalletTransactions__memo"
            >
              Test Memo
            </span>
          </div>,
        ],
        "className": "UserWalletTransactions__transaction",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Avatar
              size={40}
              username="holadweb"
            />,
            "className": "UserWalletTransactions__avatar",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "size": 40,
              "username": "holadweb",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="UserWalletTransactions__content-recipient"
              >
                <div>
                  <FormattedMessage
                    defaultMessage="Received from {username}"
                    id="received_from"
                    values={
                      Object {
                        "username": <Link
                          replace={false}
                          to="/@holadweb"
                        >
                          <span
                            className="username"
                          >
                            holadweb
                          </span>
                        </Link>,
                      }
                    }
                  />
                </div>
                <div
                  className="UserWalletTransactions__received"
                >
                  +
                  <span>
                    0 BEX
                  </span>
                </div>
              </div>,
              <span
                className="UserWalletTransactions__timestamp"
              >
                <BTooltip
                  title={
                    <span>
                      <FormattedDate
                        value="0Z"
                      />

                      <FormattedTime
                        value="0Z"
                      />
                    </span>
                  }
                >
                  <span>
                    <FormattedRelative
                      updateInterval={10000}
                      value="0Z"
                    />
                  </span>
                </BTooltip>
              </span>,
              <span
                className="UserWalletTransactions__memo"
              >
                Test Memo
              </span>,
            ],
            "className": "UserWalletTransactions__content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    <FormattedMessage
                      defaultMessage="Received from {username}"
                      id="received_from"
                      values={
                        Object {
                          "username": <Link
                            replace={false}
                            to="/@holadweb"
                          >
                            <span
                              className="username"
                            >
                              holadweb
                            </span>
                          </Link>,
                        }
                      }
                    />
                  </div>,
                  <div
                    className="UserWalletTransactions__received"
                  >
                    +
                    <span>
                      0 BEX
                    </span>
                  </div>,
                ],
                "className": "UserWalletTransactions__content-recipient",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <FormattedMessage
                      defaultMessage="Received from {username}"
                      id="received_from"
                      values={
                        Object {
                          "username": <Link
                            replace={false}
                            to="/@holadweb"
                          >
                            <span
                              className="username"
                            >
                              holadweb
                            </span>
                          </Link>,
                        }
                      }
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "defaultMessage": "Received from {username}",
                      "id": "received_from",
                      "values": Object {
                        "username": <Link
                          replace={false}
                          to="/@holadweb"
                        >
                          <span
                            className="username"
                          >
                            holadweb
                          </span>
                        </Link>,
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "+ ",
                      <span>
                        0 BEX
                      </span>,
                    ],
                    "className": "UserWalletTransactions__received",
                  },
                  "ref": null,
                  "rendered": Array [
                    "+ ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "0 BEX",
                      },
                      "ref": null,
                      "rendered": "0 BEX",
                      "type": "span",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <BTooltip
                  title={
                    <span>
                      <FormattedDate
                        value="0Z"
                      />

                      <FormattedTime
                        value="0Z"
                      />
                    </span>
                  }
                >
                  <span>
                    <FormattedRelative
                      updateInterval={10000}
                      value="0Z"
                    />
                  </span>
                </BTooltip>,
                "className": "UserWalletTransactions__timestamp",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <span>
                    <FormattedRelative
                      updateInterval={10000}
                      value="0Z"
                    />
                  </span>,
                  "title": <span>
                    <FormattedDate
                      value="0Z"
                    />

                    <FormattedTime
                      value="0Z"
                    />
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <FormattedRelative
                      updateInterval={10000}
                      value="0Z"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "updateInterval": 10000,
                      "value": "0Z",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "span",
                },
                "type": [Function],
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Test Memo",
                "className": "UserWalletTransactions__memo",
              },
              "ref": null,
              "rendered": "Test Memo",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
